service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseSubmissionValidatorServiceTest.java

```diff
@@ -0,0 +1,267 @@
+                .orderType(ImmutableList.of(OrderType.EMERGENCY_PROTECTION_ORDER))
+                .build())
+            .build();
+
+        List<String> errors = service.validateCaseDetails(caseData);
+        assertThat(errors).contains("- Enter a valid address for the contact");
+        assertThat(errors).contains("- Enter a postcode for the contact");
+    }
+
+    @Test
+    void shouldNotReturnErrorsIfMandatoryFieldsHaveBeenCompletedNotIncludingEPO() {
+        CaseData caseData = CaseData.builder()
+            .caseName("Test case")
+            .orders(Orders.builder()
+                .orderType(ImmutableList.of(OrderType.EDUCATION_SUPERVISION_ORDER))
+                .build())
+            .children(Children.builder()
+                .firstChild(Child.builder()
+                    .childName("Test")
+                    .build())
+                .additionalChildren(ImmutableList.of(
+                    Element.<Child>builder()
+                        .id(UUID.randomUUID())
+                        .value(Child.builder()
+                            .childName("Test")
+                            .build())
+                        .build()
+                ))
+                .build())
+            .applicant(Applicant.builder()
+                .name("Harry Kane")
+                .personToContact("Harry Kane")
+                .jobTitle("Judge")
+                .address(Address.builder()
+                    .addressLine1("1 Some street")
+                    .addressLine2("Some road")
+                    .postTown("some town")
+                    .postcode("BT66 7RR")
+                    .county("Some county")
+                    .country("UK")
+                    .build())
+                .telephone("02838882404")
+                .email("Harrykane@HMCTS.net")
+                .build())
+            .documents_socialWorkStatement_document(Document.builder()
+                .documentStatus("reason")
+                .build())
+            .documents_checklist_document(Document.builder()
+                .documentStatus("reason")
+                .build())
+            .documents_socialWorkAssessment_document(Document.builder()
+                .documentStatus("reason")
+                .build())
+            .documents_socialWorkCarePlan_document(Document.builder()
+                .documentStatus("reason")
+                .build())
+            .documents_socialWorkChronology_document(Document.builder()
+                .documentStatus("reason")
+                .build())
+            .documents_socialWorkEvidenceTemplate_document(Document.builder()
+                .documentStatus("reason")
+                .build())
+            .documents_threshold_document(Document.builder()
+                .documentStatus("reason")
+                .build())
+            .hearing(Hearing.builder()
+                .timeFrame("Within 18 days")
+                .build())
+            .build();
```

Comment:
so much setup, how could we make this better 🤔

Reply:
Move convo to above comment